tests/components/demo/test_notify.py

```diff
@@ -8,189 +9,198 @@
+    hass.states.async_set("sensor.temperature", 10)
+    data = {
+        notify.ATTR_MESSAGE: "{{states.sensor.temperature.state}}",
+        notify.ATTR_TITLE: "{{ states.sensor.temperature.name }}",
+    }
+    await hass.services.async_call(notify.DOMAIN, notify.SERVICE_NOTIFY, data)
+    await hass.async_block_till_done()
+    last_event = events[-1]
+    assert last_event.data[notify.ATTR_TITLE] == "temperature"
+    assert last_event.data[notify.ATTR_MESSAGE] == 10
+
+
+async def test_method_forwards_correct_data(hass, events):
+    """Test that all data from the service gets forwarded to service."""
+    await setup_notify(hass)
+    data = {
+        notify.ATTR_MESSAGE: "my message",
+        notify.ATTR_TITLE: "my title",
+        notify.ATTR_DATA: {"hello": "world"},
+    }
+    await hass.services.async_call(notify.DOMAIN, notify.SERVICE_NOTIFY, data)
+    await hass.async_block_till_done()
+    assert len(events) == 1
+    data = events[0].data
+    assert {
+        "message": "my message",
+        "title": "my title",
+        "data": {"hello": "world"},
+    } == data
+
+
+async def test_calling_notify_from_script_loaded_from_yaml_without_title(hass, events):
+    """Test if we can call a notify from a script."""
+    await setup_notify(hass)
+    step = {
+        "service": "notify.notify",
+        "data": {
+            "data": {"push": {"sound": "US-EN-Morgan-Freeman-Roommate-Is-Arriving.wav"}}
+        },
+        "data_template": {"message": "Test 123 {{ 2 + 2 }}\n"},
+    }
+    await async_setup_component(
+        hass, "script", {"script": {"test": {"sequence": step}}}
+    )
+    await hass.services.async_call("script", "test")
+    await hass.async_block_till_done()
+    assert len(events) == 1
+    assert {
+        "message": "Test 123 4",
+        "data": {"push": {"sound": "US-EN-Morgan-Freeman-Roommate-Is-Arriving.wav"}},
+    } == events[0].data
+
+
+async def test_calling_notify_from_script_loaded_from_yaml_with_title(hass, events):
+    """Test if we can call a notify from a script."""
+    await setup_notify(hass)
+    step = {
+        "service": "notify.notify",
+        "data": {
+            "data": {"push": {"sound": "US-EN-Morgan-Freeman-Roommate-Is-Arriving.wav"}}
+        },
+        "data_template": {"message": "Test 123 {{ 2 + 2 }}\n", "title": "Test"},
+    }
+    await async_setup_component(
+        hass, "script", {"script": {"test": {"sequence": step}}}
+    )
+    await hass.services.async_call("script", "test")
+    await hass.async_block_till_done()
+    assert len(events) == 1
+    assert {
+        "message": "Test 123 4",
+        "title": "Test",
+        "data": {"push": {"sound": "US-EN-Morgan-Freeman-Roommate-Is-Arriving.wav"}},
+    } == events[0].data
+
+
+async def test_targets_are_services(hass):
+    """Test that all targets are exposed as individual services."""
+    await setup_notify(hass)
+    assert hass.services.has_service("notify", "demo") is not None
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6356951)